ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -0,0 +1,962 @@
+    ls.add(highValue + dc.getHighValue() + "'");
+    ls.add(lowValue + dc.getLowValue() + "'");
+  }
+
+  public boolean checkBitVectors(ColumnStatisticsData cd) {
+    if (cd.isSetDoubleStats() && cd.getDoubleStats().isSetBitVectors()) {
+      return true;
+    }
+    if (cd.isSetBinaryStats() && cd.getBinaryStats().isSetBitVectors()) {
+      return true;
+    }
+    if (cd.isSetStringStats() && cd.getStringStats().isSetBitVectors()) {
+      return true;
+    }
+    if (cd.isSetDateStats() && cd.getDateStats().isSetBitVectors()) {
+      return true;
+    }
+    if (cd.isSetLongStats() && cd.getLongStats().isSetBitVectors()) {
+      return true;
+    }
+    if (cd.isSetDecimalStats() && cd.getDecimalStats().isSetBitVectors()) {
+      return true;
+    }
+    if (cd.isSetDoubleStats() && cd.getDoubleStats().isSetBitVectors()) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Parses the basic ColumnStatObject and returns the alter tables stmt for each individual column in a table.
+   *
+   * @param columnStatisticsData
+   * @param colName
+   * @param tblName
+   * @param dbName
+   * @return
+   */
+  public String getAlterTableStmtCol(ColumnStatisticsData columnStatisticsData, String colName, String tblName, String dbName) {
+    ST command = new ST(ALTER_TABLE_UPDATE_STATISTICS_TABLE_COLUMN);
+    command.add(DATABASE_NAME, dbName);
+    command.add(TABLE_NAME, tblName);
+    command.add(COLUMN_NAME, colName);
+    List<String> temp = new ArrayList<>();
+    addBinaryStats(columnStatisticsData, temp);
+    addLongStats(columnStatisticsData, temp);
+    addBooleanStats(columnStatisticsData, temp);
+    addStringStats(columnStatisticsData, temp);
+    addDateStats(columnStatisticsData, temp);
+    addDoubleStats(columnStatisticsData, temp);
+    addDecimalStats(columnStatisticsData, temp);
+    command.add(TBLPROPERTIES, Joiner.on(",").join(temp));
+    return command.render();
+  }
+
+  /**
+   * Parses the ColumnStatistics for all the columns in a given table and adds the alter table update
+   * statistics command for each column.
+   *
+   * @param tbl
+   */
+  public List<String> getAlterTableStmtTableStatsColsAll(Table tbl)
+      throws HiveException {
+    List<String> alterTblStmt = new ArrayList<String>();
+    List<String> accessedColumns = getTableColumnNames(tbl);
+    List<ColumnStatisticsObj> tableColumnStatistics = Hive.get().getTableColumnStatistics(tbl.getDbName(),
+        tbl.getTableName(),
+        accessedColumns,
+        true);
+    ColumnStatisticsObj[] columnStatisticsObj = tableColumnStatistics.toArray(new ColumnStatisticsObj[0]);
+    for (int i = 0; i < columnStatisticsObj.length; i++) {
+      alterTblStmt.add(getAlterTableStmtCol(columnStatisticsObj[i].getStatsData(),
+          columnStatisticsObj[i].getColName(),
+          tbl.getTableName(), tbl.getDbName()));
+      if (checkBitVectors(columnStatisticsObj[i].getStatsData())) {
+        ST command = new ST(EXIST_BIT_VECTORS);
```

Comment:
Include BIT_VECTORS as base64 encoded format for later consumption.
e.g Something along the lines of Base64.getEncoder().encodeToString(blah.blah.getBitVectors());

This way, we can enhance it later to dump the bitvectors directly into metastore. In case it is hard in this patch, feel free to create a follow up ticket and fix it.

Reply:
Will the bit-vectors byte array be of the same length every time?